sawmill_api/lib/oltp.py

```diff
@@ -38,17 +44,31 @@ def init_app(self, app):
             ENGINE = self
 
     @contextmanager
-    def get_conn(self, timeout=None):
+    def get_conn(self, timeout=None, retries=5):
         self.semaphore.acquire(timeout=timeout)
         conn = self._pool.pop()
         if conn is None:
-            conn = psycopg2.connect(
-                host=self._host,
-                port=self._port,
-                dbname=self._dbname,
-                user=self._user,
-                password=self._password,
-            )
+            for attempt in range(1, retries):
```

Comment:
This is likely to get some re-working at a point in the future. Like, having to do linear/exponential back-off in a retry loop is common enough that a general utility function will be necessary. But until that time comes, we'll live with this one-off, not great solution of doing linear back-off (aka I know this is a bit sloppy, but I don't care at the moment; working code is better than a great idea 😉).